### PR TITLE
RecorderStatus enum class

### PR DESCRIPTION
**Pull request by stellar-aria**
Creates an enum class for the RecorderStatus

### CODE REVIEW

File: src/deluge/model/sample/sample_recorder.h [hunk 1]
```diff
@@ -83,19 +88,19 @@ class SampleRecorder {
...
+	bool autoDeleteWhenDone = false;
 	bool keepingReasonsForFirstClusters;
 	uint8_t recordingNumChannels;
-	bool hadCardError;
-	bool reachedMaxFileSize;
+	bool hadCardError = false;
+	bool reachedMaxFileSize = false;
 	bool recordingExtraMargins;
 	bool pointerHeldElsewhere;
 	bool capturedTooMuch;
```
Comment (m-m-adams) [inline src/deluge/model/sample/sample_recorder.h:106]: should these be initialized too?

Comment (stellar-aria) [reply src/deluge/model/sample/sample_recorder.h:106]: probably but I don't know what they were/should be

Comment (m-m-adams) [reply src/deluge/model/sample/sample_recorder.h:106]: Yeah I think false is right, they'll all be false after a successful abort/finish recording

Comment (sapphire-arches) [inline src/deluge/model/sample/sample_recorder.h:106]:
```suggestion
	bool recordingExtraMargins = false;
	bool pointerHeldElsewhere = false;
	bool capturedTooMuch = false;
```

Comment (sapphire-arches) [reply src/deluge/model/sample/sample_recorder.h]: I pushed a fix for this, were there any other concerns?

Comment (stellar-aria) [reply src/deluge/model/sample/sample_recorder.h]: not that I know of, :shipit:

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**